renderdoc/core/resource_manager.h

```diff
@@ -40,18 +40,22 @@ using std::map;
 // used to determine if initial contents are needed and to what degree
 enum FrameRefType
 {
-  eFrameRef_Unknown,    // for the initial start of frame pipeline state - can't be marked as
-                        // written/read yet until first action.
+  eFrameRef_None,               // initial state, no reads or writes
+  eFrameRef_Clear,              // clear the entire resuorce
+  eFrameRef_Write,              // write to some subset of resource
+  eFrameRef_Read,               // read from resource
+  eFrameRef_ReadBeforeWrite,    // read followed by write
+};
 
-  // Inputs
-  eFrameRef_Read,
-  eFrameRef_Write,
+// Compose frame refs that occur in a known order.
```

Comment:
It would be helpful to ASCII-art or comment in prose here the state machine you're constructing, as reference.